test/core/common/scale_test.cpp

```diff
@@ -0,0 +1,1226 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "common/result.hpp"
+#include "common/scale.hpp"
+#include "common/scale/basic_stream.hpp"
+
+using namespace kagome;
+using namespace kagome::common;
+using namespace common::scale;
+
+/**
+ * Utility tests
+ */
+
+/**
+ * @given byte array of 3 items: 0, 1, 2
+ * @when create BasicStream wrapping this array and start to get bytes one by
+ * one
+ * @then bytes 0, 1, 2 are obtained sequencially
+ */
+TEST(Scale, basicStream) {
+  auto bytes = ByteArray{0, 1, 2};
+  auto stream = BasicStream{bytes};
+  ASSERT_EQ(stream.hasMore(3), true);
+
+  auto byte = stream.nextByte();
+  ASSERT_EQ(byte.has_value(), true);
+  ASSERT_EQ((*byte), 0);
+
+  ASSERT_EQ(stream.hasMore(2), true);
+  byte = stream.nextByte();
+  ASSERT_EQ(byte.has_value(), true);
+  ASSERT_EQ((*byte), 1);
+
+  ASSERT_EQ(stream.hasMore(1), true);
+  byte = stream.nextByte();
+  ASSERT_EQ(byte.has_value(), true);
+  ASSERT_EQ((*byte), 2);
+
+  ASSERT_EQ(stream.hasMore(1), false);
+}
+
+/**
+ * @given byte array of length 4 {1,2,3,4} and basic stream wrapping this byte
+ * array
+ * @when union array of type uint32_t reads data from stream
+ * @then its value matches expected number and its bytes correspond to source
+ * byte array
+ */
+TEST(Scale, readUnionArray) {
+  auto src = ByteArray{1, 2, 3, 4};
```

Comment:
You can create a fixture class and put such common values as this one as class variables, so that tests under that fixture would have access to it

Reply:
Ok, I will learn how to create fixtures in google tests and apply them after more urgent changes.